deeppavlov/models/vectorizers/dictionary_vectorizer.py

```diff
@@ -0,0 +1,253 @@
+import pathlib
+from collections import defaultdict
+import re
+from typing import List, Dict, Generator, Tuple, Any, AnyStr
+
+import numpy as np
+
+from sortedcontainers import SortedListWithKey
+from pymorphy2 import MorphAnalyzer
+from russian_tagsets import converters
+
+from deeppavlov.core.models.serializable import Serializable
+from deeppavlov.core.models.component import Component
+from deeppavlov.core.common.registry import register
+
+
+@register("dictionary_vectorizer")
+class DictionaryVectorizer(Serializable):
+    """
+    Transforms words into 0-1 vector of its possible tags, read from a vocabulary file.
+    The format of the vocabulary must be
+        word<TAB>tag_1<SPACE>...<SPACE>tag_k
+
+    save_path: str, path to save the vocabulary,
+    load_path: str or list of strs, path to the vocabulary(-ies),
+    min_freq: int, default=1, minimal frequency of tag to memorize this tag,
+    unk_token: str or None, unknown token to be yielded for unknown words
+    """
+    def __init__(self, save_path, load_path, min_freq=1, unk_token=None, **kwargs):
+        super().__init__(save_path, load_path, **kwargs)
+        self.min_freq = min_freq
+        self.unk_token = unk_token
+        self.load()
+
+    @property
+    def dim(self):
+        return len(self._t2i)
+
+    def save(self):
+        save_path = str(self.save_path)
+        with open(save_path, "w", encoding="utf8") as fout:
+            for word, curr_labels in sorted(self.word_tag_mapping.items()):
+                curr_labels = [self._i2t[index] for index in curr_labels]
+                curr_labels = [x for x in curr_labels if x != self.unk_token]
+                fout.write("{}\t{}".format(word, " ".join(curr_labels)))
+        return self
+
+    def load(self):
+        if isinstance(self.load_path, str):
+            load_path = pathlib.Path(self.load_path)
+            if load_path.is_dir():
+                load_path = [str(x) for x in load_path.iterdir() if x.is_file()]
+            else:
+                load_path = [str(load_path)]
+        else:
+            load_path = [str(x) for x in self.load_path]
+        labels_by_words = defaultdict(set)
+        for infile in load_path:
+            with open(infile, "r", encoding="utf8") as fin:
+                for line in fin:
+                    line = line.strip()
+                    if line.count("\t") != 1:
+                        continue
+                    word, labels = line.split("\t")
+                    labels_by_words[word].update(labels.split())
+        self._train(labels_by_words)
+        return self
+
+    def _train(self, labels_by_words : Dict):
```

Comment:
If it's trainable, should it not be an Estimator?

Reply:
Renamed to _initialize since _train does simple loading.